pymc_extras/statespace/models/structural/components/regression.py

```diff
@@ -59,15 +60,21 @@ def _handle_input_data(self, k_exog: int, state_names: list[str] | None, name) -
         return k_exog
 
     def make_symbolic_graph(self) -> None:
-        betas = self.make_and_register_variable(f"beta_{self.name}", shape=(self.k_states,))
-        regression_data = self.make_and_register_data(
-            f"data_{self.name}", shape=(None, self.k_states)
-        )
+        k_endog = self.k_endog
+        k_states = self.k_states // k_endog
+        self.k_posdef // k_endog
+
+        betas = self.make_and_register_variable(f"beta_{self.name}", shape=(k_endog, k_states))
+        regression_data = self.make_and_register_data(f"data_{self.name}", shape=(None, k_states))
 
-        self.ssm["initial_state", :] = betas
-        self.ssm["transition", :, :] = np.eye(self.k_states)
+        self.ssm["initial_state", :] = betas.reshape((1, -1)).squeeze()
+        T = np.eye(k_states)
+        self.ssm["transition", :, :] = linalg.block_diag(*[T for _ in range(k_endog)])
```

Comment:
use `pt.linalg.block_diag` instead, otherwise pytensor can't reason about the block diag